pandas/_libs/intervaltree.pxi.in

```diff
@@ -128,7 +136,12 @@ cdef class IntervalTree(IntervalMixin):
         result = Int64Vector()
         old_len = 0
         for i in range(len(target)):
-            self.root.query(result, target[i])
+            try:
+                self.root.query(result, target[i])
+            except OverflowError:
```

Comment:
Need to add these as operations like `IntervalTree[uint64].get_indexer([-1])` will raise an `OverflowError`.  See the associated tests I added.